### PR TITLE
CEDS-2066 Generate random arrival reference

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: app/uk/gov/hmrc/exports/movements/models/movements/Movement.scala [hunk 1]
```diff
@@ -27,10 +29,21 @@ case class Movement(
...
+      (JsPath \ "choice").read[MovementType] and
+      (JsPath \ "consignmentReference").read[ConsignmentReference] and
+      (JsPath \ "movementDetails").readNullable[MovementDetails] and
+      (JsPath \ "location").readNullable[Location] and
+      (JsPath \ "transport").readNullable[Transport]
+  )(Movement.apply _)
+
+  val writes: Writes[Movement] = Json.writes[Movement]
+
+  implicit val format: Format[Movement] = Format(reads, writes)
```
Comment (PatrykRudnicki) [inline app/uk/gov/hmrc/exports/movements/models/movements/Movement.scala]: Custom reads are temporary, to ignore arrival reference from frontend, will remove it when I remove the frontend screen

File: app/uk/gov/hmrc/exports/movements/models/movements/Movement.scala [hunk 2]
```diff
@@ -27,10 +29,21 @@ case class Movement(
   consignmentReference: ConsignmentReference,
   movementDetails: Option[MovementDetails] = None,
   location: Option[Location] = None,
-  arrivalReference: Option[ArrivalReference] = None,
   transport: Option[Transport] = None
 ) extends UserIdentification
 
 object Movement extends JSONResponses {
-  implicit val format = Json.format[Movement]
+  val reads: Reads[Movement] = (
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/exports/movements/models/movements/Movement.scala]: Why do we need this?

File: app/uk/gov/hmrc/exports/movements/services/ILEMapper.scala [hunk 3]
```diff
@@ -52,6 +53,11 @@ class ILEMapper @Inject()(clock: Clock) {
       case _                    => None
     }
 
+    val movementReference: Option[String] = request.choice match {
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/exports/movements/services/ILEMapper.scala]: :+1:

File: app/uk/gov/hmrc/exports/movements/models/movements/Movement.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/movements/models/movements/Movement.scala]: Yeah, you're right. Fixing...

Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/movements/models/movements/Movement.scala]: Fixed :)

**Comment by PatrykRudnicki** [comment]
test this please

**Comment by PatrykRudnicki** [comment]
PR builder is not working, not sure why, all tests runned locally and passed, coverage checked. Merging...

**Pull request merged**